#! /bin/sh /usr/share/dpatch/dpatch-run
## stable-042-3782_gtkhtml-sharp-3.14.dpatch by Tim Retout <tim@retout.co.uk>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Patch from upstream stable branch for allowing to build against
## DP: gtkhtml-sharp 3.14.

@DPATCH@
commit 7c8592fd781317d0151a101edfb3ffbadac4839c
Author: sdelcroix <sdelcroix@772769d8-d925-0410-89eb-a6ffa0d40526>
Date:   Tue Mar 25 07:42:46 2008 +0000

    backporting -r3781 to the stable branch
    
    git-svn-id: http://svn.gnome.org/svn/f-spot/branches/FSPOT_0_4_2_STABLE@3782 772769d8-d925-0410-89eb-a6ffa0d40526

diff --git a/ChangeLog b/ChangeLog
index de13645..5ece9ed 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,8 @@
+2008-03-25  Stephane Delcroix  <sdelcroix@novell.com>
+
+	* configure.in:
+	* src/InfoDisplay.cs: allow building on gnome > 2.20. Fixes bgo #524004
+
 2008-03-12  Stephane Delcroix  <sdelcroix@novell.com>
 
 	* Updater.cs: fixes the dp updating v8 where version names are int.
diff --git a/configure.in b/configure.in
index 23f8579..86be953 100644
--- a/configure.in
+++ b/configure.in
@@ -73,22 +73,23 @@ PKG_CHECK_MODULES(F, libgnome-2.0 >= $LIBGNOME_REQUIRED libgnomeui-2.0 >= $LIBGN
 AC_SUBST(F_CFLAGS)
 AC_SUBST(F_LIBS)
 
+CSC_DEFINES=""
 PKG_CHECK_MODULES(GNOME_SHARP, gnome-sharp-2.0 >= 2.8)
 if pkg-config --atleast-version=2.19.90 gnome-sharp; then
 	PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-3.14 >= 2.19.90)
-	GTKHTML_SHARP_PKG="-pkg:gtkhtml-sharp-3.14"
+ 	GTKHTML_SHARP_PKG="-pkg:gtkhtml-sharp-3.14 -pkg:gnome-sharp-2.0"
+ 	CSC_DEFINES="$CSC_DEFINES -d:GTKHTML_3_14"
 else
 	PKG_CHECK_MODULES(GTKHTML_SHARP, gtkhtml-sharp-2.0 >= 2.8)
 	GTKHTML_SHARP_PKG="-pkg:gtkhtml-sharp-2.0"
 fi
 AC_SUBST(GTKHTML_SHARP_PKG)
 
-CSC_DEFINES=""
 if pkg-config --atleast-version=2.10 gtk-sharp-2.0; then
-   CSC_DEFINES="-d:GTK_2_10"
+   CSC_DEFINES="$CSC_DEFINES -d:GTK_2_10"
 fi
 if pkg-config --atleast-version=2.11 gtk-sharp-2.0; then
-   CSC_DEFINES="-d:GTK_2_11"
+   CSC_DEFINES="$CSC_DEFINES -d:GTK_2_11"
 fi
 
 dnl - Choose PreferenceBackend (default to gconf)
diff --git a/src/InfoDisplay.cs b/src/InfoDisplay.cs
index 8a621d5..ed74517 100644
--- a/src/InfoDisplay.cs
+++ b/src/InfoDisplay.cs
@@ -148,7 +148,11 @@ namespace FSpot {
 
 		private void Update ()
 		{
+#if GTKHTML_3_14
+			Gtk.HTMLStream stream = this.Begin (null, "text/html; charset=utf-8", Gtk.HTMLBeginFlags.KeepScroll);
+#else
 			Gtk.HTMLStream stream = this.Begin (null, "text/html; charset=utf-8", Gtk.HTMLBeginFlags.Scroll);
+#endif
 			
 			string bg = Color (this.Style.Background (Gtk.StateType.Active));
 			string fg = Color (this.Style.Foreground (Gtk.StateType.Active));
